components/Modal/ExportRecipe.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+
+class ExportRecipe extends React.Component {
+
+  componentDidMount(){
+    $(this.refs.modal).modal('show');
+    $(this.refs.modal).on('hidden.bs.modal', this.props.handleHideModalExport);
+  }
+
+  handleCopy(){
+    this.refs.recipe_contents_text.select();
+    document.execCommand('copy');
+  }
+
+  render() {
+    return (
+      <div
+        className="modal fade"
+        id="cmpsr-modal-export"
+        ref="modal"
+        tabIndex="-1"
+        role="dialog"
+        aria-labelledby="myModalLabel"
+        aria-hidden="true"
+      >
+        <div className="modal-dialog">
+          <div className="modal-content">
+            <div className="modal-header">
+              <button
+                type="button"
+                className="close"
+                data-dismiss="modal"
+              >
+                <span className="pficon pficon-close"></span>
+              </button>
+              <h4 className="modal-title" id="myModalLabel">Export Recipe</h4>
+            </div>
+            <div className="modal-body">
+              <form className="form-horizontal" onKeyPress={(e) => this.handleEnterKey(e)}>
+                <div className="form-group">
+                  <label
+                    className="col-sm-3 control-label"
+                  >Recipe</label>
+                  <div className="col-sm-9">
+                    <p className="form-control-static">{this.props.recipe}</p>
+                  </div>
+                </div>
+                <div className="form-group">
+                  <label
+                    className="col-sm-3 control-label"
+                    htmlFor="textInput-modal-markup"
+                  >Export as</label>
+                  <div className="col-sm-9">
+                    <select className="form-control">
+                      <option>Text</option>
+                    </select>
+                  </div>
+                </div>
+                <div className="form-group">
+                  <label
+                    className="col-sm-3 control-label"
+                    htmlFor="textInput2-modal-markup"
+                  >Contents</label>
+                  { this.props.contents &&
+                    <div className="col-sm-9">
+                      <textarea
+                        readOnly
+                        id="textInput2-modal-markup"
+                        ref="recipe_contents_text"
```

Comment:
According to [React Docs](https://facebook.github.io/react/docs/refs-and-the-dom.html#legacy-api-string-refs), this string ref is a Legacy API, it should be written with callback pattern instead.